File: src/main/scala/com/yanns/scala/example1/version3/InvoiceController.scala
package com.yanns.scala.example1.version3

import java.util.Date
import com.yanns.scala.example1.version1.InvoiceCreator

class InvoiceController {

   val invoiceCreator = new InvoiceCreator()

   def payInvoice {

     val newInvoiceId = invoiceCreator.createInvoiceName(
       invoiceDate = new Date,
       invoiceId = "45AE-45F",
       userId = "872346",
       firmId = "abc-234",
       directInvoice = true,
       firmInvoice = false,
       electronic = false
     )

     // ...
   }

 }

File: src/main/java/com/yanns/java/example1/version3/InvoiceController.java
package com.yanns.java.example1.version3;

import java.util.Date;

public class InvoiceController {

    InvoiceCreator invoiceCreator = new InvoiceCreator();

    public void payInvoice() {
        NewInvoiceNameCommand newInvoiceNameCommand = new NewInvoiceNameCommand(new Date(), "45AE-45F", "872346", "abc-234", true, false, false);

        String newInvoiceId = invoiceCreator.createInvoiceName(newInvoiceNameCommand);

        // ...
    }
}
